nx-X11/programs/Xserver/hw/nxagent/Events.c

```diff
@@ -3647,21 +3649,28 @@ int nxagentHandleReparentNotify(XEvent* X)
       {
         result = XQueryTree(nxagentDisplay, w, &rootReturn, &parentReturn,
                                 &childrenReturn, &nchildrenReturn);
-    
-        if (parentReturn == rootReturn || parentReturn == 0 || result == 0)
+
+        if (childrenReturn)
         {
-          break;
+          XFree(childrenReturn);
+          childrenReturn = NULL;
         }
 
-        if (result == 1 && childrenReturn != NULL)
+        if (parentReturn == rootReturn || parentReturn == 0 || result == 0)
         {
-          XFree(childrenReturn);
+          break;
         }
-    
+
         w = parentReturn;
       }
       while (True);
 
+      if (childrenReturn)
+      {
+        XFree(childrenReturn);
+        childrenReturn = NULL;
+      }
+
```

Comment:
I think this check and Xfree is not needed.  childrenReturn is always free'd in the loop, there is no break before it. So it will be always NULL here.